### PR TITLE
Refactor create service hub context method

### PR DESCRIPTION
**Pull request by Y-Sindo**

### CODE REVIEW

File: src/Microsoft.Azure.SignalR.Management/DependencyInjectionExtensions.cs [hunk 1]
```diff
@@ -38,6 +38,14 @@ public static IServiceCollection AddSignalRServiceManager(this IServiceCollectio
             return services.AddSignalRServiceCore();
         }
 
+        public static IServiceCollection AddHub(this IServiceCollection services, string hubName)
+        {
+            return services.AddSingleton<IServiceConnectionContainer>(sp =>
+sp.GetRequiredService<MultiEndpointConnectionContainerFactory>().CreateAndStart(hubName))
```
Comment (vwxyzh) [inline src/Microsoft.Azure.SignalR.Management/DependencyInjectionExtensions.cs:44]: indent

File: src/Microsoft.Azure.SignalR.Management/DependencyInjectionExtensions.cs [hunk 2]
```diff
@@ -38,6 +38,14 @@ public static IServiceCollection AddSignalRServiceManager(this IServiceCollectio
...
+                .AddSingleton(sp => sp.GetRequiredService<ServiceHubLifetimeManagerFactory>().Create(hubName))
+                .AddSingleton(sp => (HubLifetimeManager<Hub>)sp.GetRequiredService<IServiceHubLifetimeManager>()).AddSingleton<IServiceHubContext>(sp => ActivatorUtilities.CreateInstance<ServiceHubContext>(sp, hubName));
```
Comment (wanlwanl) [inline src/Microsoft.Azure.SignalR.Management/DependencyInjectionExtensions.cs:46]: new line for the last AddSingleton

File: src/Microsoft.Azure.SignalR.Management/HubInstanceFactories/MultiEndpointConnectionContainerFactory.cs [hunk 3]
```diff
@@ -27,31 +24,16 @@ public MultiEndpointConnectionContainerFactory(IServiceConnectionFactory connect
...
-                endpoint => new WeakServiceConnectionContainer(_connectionFactory, _connectionCount, endpoint, loggerFactory.CreateLogger<WeakServiceConnectionContainer>()),
+                endpoint => new WeakServiceConnectionContainer(_connectionFactory, _connectionCount, endpoint, _loggerFactory.CreateLogger<WeakServiceConnectionContainer>()),
                 _endpointManager,
                 _router,
-                loggerFactory);
-                container.StartAsync();
-                return container;
-            }, true);
+                _loggerFactory);
+            container.StartAsync();
```
Comment (wanlwanl) [inline src/Microsoft.Azure.SignalR.Management/HubInstanceFactories/MultiEndpointConnectionContainerFactory.cs:35]: so we don't know when the container is started or failed to start when cx is already sending messages?

File: src/Microsoft.Azure.SignalR.Management/HubInstanceFactories/MultiEndpointConnectionContainerFactory.cs [hunk 4]
```diff
@@ -27,31 +24,16 @@ public MultiEndpointConnectionContainerFactory(IServiceConnectionFactory connect
             _router = router;
         }
 
-        public MultiEndpointServiceConnectionContainer GetOrCreate(string hubName, ILoggerFactory loggerFactoryPerHub = null)
+        public MultiEndpointServiceConnectionContainer CreateAndStart(string hubName)
```
Comment (wanlwanl) [inline src/Microsoft.Azure.SignalR.Management/HubInstanceFactories/MultiEndpointConnectionContainerFactory.cs:27]: naming: ConnectAsync? e,g, Microsoft.Azure.SignalR.ConnectionFactory

File: src/Microsoft.Azure.SignalR.Management/HubInstanceFactories/MultiEndpointConnectionContainerFactory.cs [hunk 5]
```diff
@@ -27,31 +24,16 @@ public MultiEndpointConnectionContainerFactory(IServiceConnectionFactory connect
...
         {
-            return _hubConnectionContainers.GetOrAdd(hubName, (hubName) => Create(hubName, loggerFactoryPerHub)).Value;
```
Comment (wanlwanl) [inline src/Microsoft.Azure.SignalR.Management/HubInstanceFactories/MultiEndpointConnectionContainerFactory.cs:32]: why _hubConnectionContainers should be removed?

Comment (Y-Sindo) [reply src/Microsoft.Azure.SignalR.Management/HubInstanceFactories/MultiEndpointConnectionContainerFactory.cs:32]: We don't need to reuse the container now.

File: src/Microsoft.Azure.SignalR.Management/HubInstanceFactories/MultiEndpointConnectionContainerFactory.cs (hunk 3 above)
Comment (Y-Sindo) [reply src/Microsoft.Azure.SignalR.Management/HubInstanceFactories/MultiEndpointConnectionContainerFactory.cs:35]: We wait [here](https://github.com/Azure/azure-signalr/pull/1184/files#diff-8c62721fa2e48aa7b5a9d16291699f40f45fa98b2e436eaf3e757a2e2356e45eR50)

File: src/Microsoft.Azure.SignalR.Management/HubInstanceFactories/MultiEndpointConnectionContainerFactory.cs (hunk 4 above)
Comment (Y-Sindo) [reply src/Microsoft.Azure.SignalR.Management/HubInstanceFactories/MultiEndpointConnectionContainerFactory.cs:27]: It is a container factory not a connection factory?

Comment (wanlwanl) [reply src/Microsoft.Azure.SignalR.Management/HubInstanceFactories/MultiEndpointConnectionContainerFactory.cs:27]: connection vs connectionContainer: only 1 vs many?

File: src/Microsoft.Azure.SignalR.Management/HubInstanceFactories/MultiEndpointConnectionContainerFactory.cs (hunk 3 above)
Comment (wanlwanl) [reply src/Microsoft.Azure.SignalR.Management/HubInstanceFactories/MultiEndpointConnectionContainerFactory.cs:35]: move the startAsync to the line before await can be clearer?
rename the method to Create()?

**Review by wanlwanl — APPROVED** [review]

**Pull request merged**